wee_slack.py

```diff
@@ -2454,6 +2456,46 @@ def scrolled_cb(signal, sig_type, data):
         pass
     return w.WEECHAT_RC_OK
 
+def convert_attributes(text):
+    if not config.convert_attributes: return text
+    matches = {
+        "*": "bold",
+        "_": "italic",
+        "~": "underline"    # TODO: change to unicode strikethrough
+    }
+    # Find any of the characters within ` and make sure they don't match
+    codes = list(re.finditer("(`.+?`)", text))
+    effect = ""
+    def filter_match(match):
+        # Check if within a code range
+        for code in codes:
+            if match.start() >= code.start() and match.start() <= code.end():
+                # Don't substitute if within code range
+                return ("{}"*6).format(
+                    match.group(1),
+                    match.group(2),
+                    match.group(3),
+                    match.group(4),
+                    match.group(5),
+                    match.group(6)
+                )
+        # Perform the substitute
+        return ("{}"*6).format(
+            match.group(1),
+            w.color(effect),
+            match.group(3),
+            match.group(4),
+            w.color("-{}".format(effect)),
+            match.group(6)
+        )
+    regex_match = "(^|\s)({0})([^{0}])(.*?)(\\2)($|\W)"
+    for key in matches:
+        effect = matches[key]
+        escaped_key = key.replace("", "\\")[:-1]
+        regex = re.compile(regex_match.format(escaped_key))
```

Comment:
If you compile with re.VERBOSE, you can add comments to the regex. Also, regex strings should start `r"`, but that's a nit.